CANScience.c

```diff
@@ -1,56 +1,59 @@
-	int16_t PWMSet)
-{
-	packetToAssemble->id = ConstructCANID(PRIO_MOTOR_UNIT_PWM_DIR_SET, targetGroup, targetSerial);
-	packetToAssemble->dlc = 4;
-	int nextByte = WritePacketIDOnly(packetToAssemble->data, ID_MOTOR_UNIT_PWM_DIR_SET);
-	PackShortIntoDataMSBFirst(packetToAssemble->data, PWMSet, nextByte);
-	packetToAssemble->data[nextByte + 2] = motor;
-}
-
-void AssembleScienceServoPacket(CANPacket *packetToAssemble,
-	uint8_t targetGroup,
-	uint8_t targetSerial,
-	uint8_t servo,
-	uint8_t degrees)
-{
-	packetToAssemble->id = ConstructCANID(PACKET_PRIORITY_NORMAL, targetGroup, targetSerial);
-	packetToAssemble->dlc = 3;
-	WritePacketIDOnly(packetToAssemble->data, ID_SCIENCE_SERVO_SET);
-	packetToAssemble->data[1] = servo;
-	packetToAssemble->data[2] = degrees;
-}
-
-int16_t GetScienceMotorPWMFromPacket(CANPacket *packet){
-	return DecodeBytesToIntMSBFirst(packet->data, 1, 2);
-}
-
-uint8_t GetScienceServoAngleFromPacket(CANPacket *packet){
-	return packet->data[2];
-}
-
-uint8_t GetScienceMotorIDFromPacket(CANPacket *packet){
-	return packet->data[3];
-}
-
-uint8_t GetScienceServoIDFromPacket(CANPacket *packet){
-	return packet->data[1];
-}
-
-
+#include "CANPacket.h"
+#include "CANCommon.h"
+#include "CANScience.h"
+#include "CANMotorUnit.h"
+
+void AssembleScienceLazySusanPosSetPacket(CANPacket *packetToAssemble,
+										  uint8_t targetDeviceGroup,
+										  uint8_t targetDeviceSerial,
+										  uint8_t position)
+{
+	packetToAssemble->id = ConstructCANID(PACKET_PRIORITY_NORMAL, targetDeviceGroup, targetDeviceSerial);
+	packetToAssemble->dlc = 2;
+	WritePacketIDOnly(packetToAssemble->data, ID_SCIENCE_LAZY_SUSAN_POS_SET);
+	packetToAssemble->data[1] = position;
+}
+
+void AssembleScienceServoPacket(CANPacket *packetToAssemble,
+	uint8_t targetGroup,
+	uint8_t targetSerial,
+	uint8_t servo,
+	uint8_t degrees)
+{
+	packetToAssemble->id = ConstructCANID(PACKET_PRIORITY_NORMAL, targetGroup, targetSerial);
+	packetToAssemble->dlc = 3;
+	WritePacketIDOnly(packetToAssemble->data, ID_SCIENCE_SERVO_SET);
+	packetToAssemble->data[1] = servo;
+	packetToAssemble->data[2] = degrees;
+}
+
+void AssembleScienceContServoPowerSetPacket(CANPacket *packetToAssemble,
+											uint8_t targetDeviceGroup,
+											uint8_t targetDeviceSerial,
+											uint8_t servo,
+											int8_t power)
+{
+	packetToAssemble->id = ConstructCANID(PACKET_PRIORITY_NORMAL, targetDeviceGroup, targetDeviceSerial);
+	packetToAssemble->dlc = 3;
+	WritePacketIDOnly(packetToAssemble->data, ID_SCIENCE_CONT_SERVO_POWER_SET);
+	packetToAssemble->data[1] = servo;
+	packetToAssemble->data[2] = power;
+}
+
+int8_t GetScienceContServoPowerFromPacket(CANPacket *packet) {
+	return packet->data[2];
+}
+
+uint8_t GetScienceServoAngleFromPacket(CANPacket *packet){
+	return packet->data[2];
+}
+
+uint8_t GetScienceLazySusanPosFromPacket(CANPacket *packet) {
+    return packet->data[1];
+}
+
+uint8_t GetScienceServoIDFromPacket(CANPacket *packet){
+	return packet->data[1];
+}
```

Comment:
Can we make the parameter type `const CANPacket *`? No reason to not have a `const` here.

Reply:
This should honestly be done consistently throughout the entire HindsightCAN library. None of the other CAN* files use `const` in this case either